libraries/c_sdk/standard/mqtt/src/iot_mqtt_managed_function_wrapper.c

```diff
@@ -74,3 +74,159 @@ IotMqttError_t _IotMqtt_managedDisconnect( IotMqttConnection_t mqttConnection )
 }
 
 /*-----------------------------------------------------------*/
+
+IotMqttError_t _IotMqtt_managedSubscribe( IotMqttConnection_t mqttConnection,
+                                          _mqttOperation_t * pSubscriptionOperation,
+                                          const IotMqttSubscription_t * pSubscriptionList,
+                                          size_t subscriptionCount )
+{
+    int8_t contextIndex = -1;
+    IotMqttError_t status = IOT_MQTT_BAD_PARAMETER;
+    /* Initializing MQTT Status. */
+    MQTTStatus_t managedMqttStatus = MQTTBadParameter;
+    uint16_t packetId = 0;
+    MQTTSubscribeInfo_t * subscriptionList = IotMqtt_MallocMessage( sizeof( MQTTSubscribeInfo_t ) * subscriptionCount );
+
+    IotMqtt_Assert( mqttConnection != NULL );
+    IotMqtt_Assert( pSubscriptionOperation != NULL );
+    IotMqtt_Assert( pSubscriptionList != NULL );
+
+    /* Getting MQTT Context for the specified MQTT Connection. */
+    contextIndex = _IotMqtt_getContextIndexFromConnection( mqttConnection );
+
+    if( contextIndex >= 0 )
+    {
+        /* Generating the packet id for SUBSCRIBE packet. */
+        packetId = _nextPacketIdentifier();
```

Comment:
We would be able to use the packet Id generation logic from MQTT LTS library instead of this private function.